File: sdk/flutter/probe_flutter/lib/src/probe_plugin.dart
/// A Probe plugin provides a specific inspection capability.
abstract class ProbePlugin {
  String get id;
  String get displayName;

  void onAttach(ProbeHost host);
  void onDetach();
}

/// The host provides transport access to plugins.
abstract class ProbeHost {
  bool get isConnected;
  void send(String pluginId, Map<String, dynamic> payload);
}

File: sdk/ios/Sources/ProbeCore/ProbePlugin.swift
/// A Probe plugin provides a specific inspection capability (network, db, prefs, …).
public protocol ProbePlugin: AnyObject {
    var id: String { get }
    var displayName: String { get }

    func onAttach(host: any ProbeHost)
    func onDetach()
}

/// The host provides transport access to plugins.
public protocol ProbeHost: AnyObject {
    var isConnected: Bool { get }
    func send(pluginId: String, payload: [String: Any?])
}
